src/components/Tool/FixPhoneNumbers/Contact.test.tsx

```diff
@@ -99,46 +92,62 @@ describe('FixPhoneNumbers-Contact', () => {
     ) as HTMLInputElement;
     const addButton = getByTestId('addButton-testid');
 
-    userEvent.type(addInput, '123');
-    expect(addInput.value).toBe('123');
+    userEvent.type(addInput, '1');
+
+    await waitFor(() => {
+      expect(setValuesMock).toHaveBeenCalledWith({
+        people: [
+          {
+            ...testData,
+            isNewPhoneNumber: true,
+            newPhoneNumber: '1',
+          },
+        ],
+      });
+    });
+
     userEvent.click(addButton);
     expect(addInput.value).toBe('');
   });
 
-  it('should call mock functions', () => {
-    const handleChangeMock = jest.fn();
-    const handleDeleteModalOpenMock = jest.fn();
-    const handleAddMock = jest.fn();
-    const handleChangePrimaryMock = jest.fn();
-    const updatePhoneNumber = jest.fn();
-
+  it('should call mock functions', async () => {
     const { getByTestId } = render(
       <ThemeProvider theme={theme}>
         <TestWrapper>
           <Contact
-            name={testData.name}
-            key={testData.name}
-            personId={testData.id}
-            numbers={testData.numbers}
-            toDelete={[]}
-            handleChange={handleChangeMock}
+            key={testData.id}
+            person={testData}
+            personIndex={0}
             handleDelete={handleDeleteModalOpenMock}
-            handleAdd={handleAddMock}
-            handleChangePrimary={handleChangePrimaryMock}
             setContactFocus={setContactFocus}
-            avatar={testData.avatar}
             handleUpdate={updatePhoneNumber}
+            errors={errors}
+            values={{ people: [testData] }}
+            setValues={setValuesMock}
           />
         </TestWrapper>
       </ThemeProvider>,
     );
 
     const firstInput = getByTestId('textfield-testid-0') as HTMLInputElement;
     expect(firstInput.value).toBe('3533895895');
-    userEvent.type(firstInput, '123');
-    expect(handleChangeMock).toHaveBeenCalled();
-    userEvent.click(getByTestId('starOutlineIcon-testid-1'));
-    expect(handleChangePrimaryMock).toHaveBeenCalled();
+    userEvent.type(firstInput, '1');
+
+    await waitFor(() => {
+      expect(setValuesMock).toHaveBeenCalledWith({
+        people: [
+          {
+            ...testData,
+            phoneNumbers: {
+              nodes: [
+                { ...testData.phoneNumbers.nodes[0], number: '35338958951' },
+                { ...testData.phoneNumbers.nodes[1] },
```

Comment:
You shouldn't need to clone this node.

```suggestion
                testData.phoneNumbers.nodes[1],
```